pinot-connectors/pinot-spark-connector/pom.xml

```diff
@@ -199,19 +186,25 @@
             </executions>
           </plugin>
           <plugin>
-            <artifactId>maven-assembly-plugin</artifactId>
-            <configuration>
-              <descriptorRefs>
-                <descriptorRef>jar-with-dependencies</descriptorRef>
-              </descriptorRefs>
-            </configuration>
+            <artifactId>maven-shade-plugin</artifactId>
             <executions>
               <execution>
-                <id>assemble-all</id>
                 <phase>package</phase>
                 <goals>
-                  <goal>single</goal>
+                  <goal>shade</goal>
                 </goals>
+                <configuration>
+                  <relocations>
+                    <relocation>
+                      <pattern>com</pattern>
+                      <shadedPattern>repackaged.com</shadedPattern>
```

Comment:
Consider follow the pattern here for presto, with your shadeBase parameter?
https://github.com/apache/pinot/blob/master/pinot-connectors/prestodb-pinot-dependencies/presto-pinot-driver/pom.xml#L39